packages/fxa-auth-server/lib/routes/emails.js

```diff
@@ -196,100 +193,94 @@ module.exports = (
+                emails = [userEmail];
+                return true;
+              }
+            });
+
+            // This user is attempting to verify a secondary email that doesn't belong to the account.
+            if (emails.length === 0) {
+              throw error.cannotResendEmailCodeToUnownedEmail();
             }
 
-            const mailerOpts = {
-              code,
-              deviceId: sessionToken.deviceId,
-              flowId,
-              flowBeginTime,
-              service,
-              ip,
-              location: geoData.location,
-              timeZone: geoData.timeZone,
-              timestamp: Date.now(),
-              redirectTo: request.payload.redirectTo,
-              resume: request.payload.resume,
-              acceptLanguage: request.app.acceptLanguage,
-              uaBrowser: sessionToken.uaBrowser,
-              uaBrowserVersion: sessionToken.uaBrowserVersion,
-              uaOS: sessionToken.uaOS,
-              uaOSVersion: sessionToken.uaOSVersion,
-              uaDeviceType: sessionToken.uaDeviceType,
-              uid: sessionToken.uid,
-              style,
-            };
-
-            return verifyFunction(emails, sessionToken, mailerOpts).then(() =>
-              request.emitMetricsEvent(`email.${event}.resent`)
-            );
-          })
-          .then(() => {
-            return {};
-          });
+            return true;
+          } else if (sessionToken.tokenVerificationId) {
+            emails = emailData;
+            code = sessionToken.tokenVerificationId;
 
-        function setVerifyCode() {
-          return db.accountEmails(sessionToken.uid).then(emailData => {
-            if (email) {
-              // If an email address is specified in payload, this is a request to verify
-              // a secondary email. This should return the corresponding email code for verification.
-              let emailVerified = false;
-              emailData.some(userEmail => {
-                if (userEmail.normalizedEmail === email.toLowerCase()) {
-                  code = userEmail.emailCode;
-                  emailVerified = userEmail.isVerified;
-                  emails = [userEmail];
-                  return true;
-                }
-              });
+            // Check to see if this account has a verified TOTP token. If so, then it should
+            // not be allowed to bypass TOTP requirement by sending a sign-in confirmation email.
+            try {
+              const result = await db.totpToken(sessionToken.uid);
 
-              // This user is attempting to verify a secondary email that doesn't belong to the account.
-              if (emails.length === 0) {
-                throw error.cannotResendEmailCodeToUnownedEmail();
+              if (result && result.verified && result.enabled) {
+                return false;
               }
-
-              // Don't resend code for already verified emails
-              if (emailVerified) {
```

Comment:
The only place that has touched this variable is in a block that returns before this line can run, I stared at it for a few minutes as well before grokking it. I'm guessing it was leftover from some change before?

Reply:
Not sure if was intentional, but it was certainly a mistake by me.  `emailVerified` was assigned as a side effect of building `emails`, which is either empty or has only one element.  I'll make some changes so that we'll return `foundEmail.isVerified` here.

Reply:
Err I meant `!foundEmail.isVerified` since we don't want to send email if the address has been verified.